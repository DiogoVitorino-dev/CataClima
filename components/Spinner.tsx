import { ActivityIndicator, useColorScheme } from "react-native";
import { View } from "./Themed";
import Colors from "../constants/Colors";

export default function Spinner() {
    const colorScheme = useColorScheme()

    return (
        <View style={{
            flex:1,
            justifyContent:'center',
            alignItems:'center'}}
        >
            <ActivityIndicator 
            size="large" 
            color={Colors[colorScheme ?? 'light'].icon} />
        </View>
    )
    
}