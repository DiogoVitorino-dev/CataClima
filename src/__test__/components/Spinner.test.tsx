import {describe,it, expect, afterEach} from '@jest/globals';
import { cleanup } from '@testing-library/react-native';
import renderer from 'react-test-renderer'
import Spinner from '../../components/Spinner';

describe('Spinner component', () => {
  afterEach(cleanup)

  it('should renders correctly', () => {    
      const tree = renderer.create(<Spinner />).toJSON();
      expect(tree).toMatchSnapshot()    
  })
})