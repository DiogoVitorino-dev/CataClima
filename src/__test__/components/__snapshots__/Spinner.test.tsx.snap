// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spinner component should renders correctly 1`] = `
<View
  style={
    [
      {
        "backgroundColor": "#fff",
      },
      {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      },
    ]
  }
>
  <Image
    source={1}
    style={
      {
        "height": 300,
        "width": 300,
      }
    }
  />
</View>
`;
